contributor-docs/adrs/adr-022-css-layer-names.md

```diff
@@ -0,0 +1,84 @@
+# CSS Layer Names
+
+## Status
+
+| Stage    | Status      |
+| -------- | ----------- |
+| Approved | 🚧          |
+| Adopted  | <!-- ✅ --> |
+
+## Context
+
+As Primer begins making use of CSS Layers, we need to define the named groups
+and order of the layers that we create. This will help to ensure a consistent
+ordering of layers and make sure that our naming is consistent across libraries.
+
+We have several areas that may be helpful to define explicit layers for:
+
+- Primitives
+- Octicons
+- Components
+- Resets
+- Utilities
+- Product
+
+This ADR will determine the naming convention for CSS Layers and the order in
+which they should be applied.
+
+## Decision
+
+Below is the list of CSS layers that we will use in Primer. They appear in the
+order in which they should be applied.
+
+- `base`: base styles that are applied to the document and the root element. This includes things like resets, typography, and
+  global styles.
+- `theme`: styles that are applied for theming. This layer includes `@primer/primitives`
+- `icons`: styles that are applied to octicons
+- `components`: styles that are applied to components
+- `utilities`: an optional layer for utility classes in order to take precedence over earlier styles
```

Comment:
Non-blocking nitpick: I see the library mentioned (and used) below, but not in this list. Might be helpful to add it here after `base` or `theme`.